File: Leetcode/JumpGameII/DP.cpp
class Solution {
public:
    int jump(int A[], int n) {
        if(n == 1)  return 0;
        if(A[0] > n)   return 1;
        vector<int> f(n, INT_MAX);
        f[0] = 0;
        int i = 1;
        while(i < n){
            int j = i - 1;
            while(j >= 0){
                if((j + A[j]) >= i){
                    f[i] = min(f[i], f[j]);
                }
                j--;
            }
            f[i]++;
            i++;
        }
        return f[n - 1];
    }
};

File: Leetcode/JumpGameII/DP.py
def jump(self, nums):
  opt = []
  for i in range(0, len(nums)):
     opt.append(sys.maxint)

  opt[0] = 0
  for i in range(1, len(nums)):
    for j in range(0, i):
        if nums[j] + j >= i:
          opt[i] = min(opt[j] + 1, opt[i])
  return opt[len(nums) - 1]
